src/main/java/org/sonatype/repository/conan/internal/metadata/ConanUrlIndexer.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2017-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.repository.conan.internal.metadata;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.sonatype.goodies.common.ComponentSupport;
+import org.sonatype.nexus.repository.Repository;
+import org.sonatype.nexus.repository.storage.Asset;
+import org.sonatype.nexus.repository.storage.Bucket;
+import org.sonatype.nexus.repository.storage.StorageFacet;
+import org.sonatype.nexus.repository.storage.StorageTx;
+import org.sonatype.nexus.repository.storage.TempBlob;
+import org.sonatype.nexus.repository.transaction.TransactionalStoreMetadata;
+import org.sonatype.nexus.repository.view.Content;
+import org.sonatype.nexus.repository.view.payloads.StringPayload;
+import org.sonatype.nexus.transaction.UnitOfWork;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Supplier;
+
+import static java.nio.charset.Charset.defaultCharset;
+import static java.util.Collections.emptyMap;
+import static org.sonatype.nexus.repository.storage.AssetEntityAdapter.P_ASSET_KIND;
+import static org.sonatype.nexus.repository.view.Content.maintainLastModified;
+import static org.sonatype.nexus.repository.view.ContentTypes.APPLICATION_JSON;
+import static org.sonatype.repository.conan.internal.AssetKind.CONAN_INDEX;
+import static org.sonatype.repository.conan.internal.proxy.ConanProxyHelper.HASH_ALGORITHMS;
+import static org.sonatype.repository.conan.internal.proxy.ConanProxyHelper.findAsset;
+import static org.sonatype.repository.conan.internal.proxy.ConanProxyHelper.toContent;
+
+/**
+ * download_url files contain absolute paths to each asset
+ *
+ * This class removes the absolute address so as to redirect back to this repository
+ *
+ * @since 3.conan
+ */
+@Singleton
+@Named
+public class ConanUrlIndexer
```

Comment:
Question: The branch I have checked out has both a ConanAbsoluteUrlIndexer and ConanUrlIndexer, is that something that has gone wrong with my local copy or should one of them be removed?

Reply:
I thin that is your local branch, at one point I renamed and there was a conflict and I accidentally pushed both files. The latest should only have ConanUrlIndexer